packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -0,0 +1,129 @@
+    // plugin will skip the "light" flavors because their data can be gotten
+    // with the dark flavor
+    if (imageSet) {
+      const darkFlavorContents = (await flavor.getContents()) as string; // .svg asset's getContents() returns a string (the svg as a string)
+      const lightFlavor = icon.flavors.get(imageSet);
+      let lightFlavorContents;
+      if (lightFlavor) {
+        lightFlavorContents = (await lightFlavor.getContents()) as string;
+
+      }
+
+    const lightToken = params.lightToken;
+    const darkToken = params.darkToken;
+
+
+    const lightSvgo = new Svgo({
+      plugins: [
+        {
+          addAttributesToSVGElement: {
+            attributes: [{display: lightToken || "var(--svg-light-display)"}]
+          }
+        }
+      ],
+      js2svg: { pretty: true, indent: 2 }
+    });
+
+    const darkSvgo = new Svgo({
+      plugins: [
+        {
+          addAttributesToSVGElement: {
+            attributes: [{display: darkToken || "var(--svg-dark-display)"}]
+          }
+        }
+      ],
+      js2svg: { pretty: true, indent: 2 }
+    });
+
+      if (lightFlavorContents){
+        // create optimized light/dark svg assets
+        const lightAsset = await lightSvgo.optimize(lightFlavorContents); //returns a string
+        const darkAsset = await darkSvgo.optimize(darkFlavorContents);
+
+        // Create SVG PARENT
+        const DOM = new DOMImplementation();
+        const doctype = DOM.createDocumentType(
+          'svg',
+          '-//W3C//DTD SVG 1.1//EN',
+          'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'
+        );
+        const SVG_NS = 'http://www.w3.org/2000/svg';
+
+        // Create an SVG Document and set its doctype
+        const DOCUMENT = DOM.createDocument(SVG_NS, 'svg', doctype);
+
+        // Create SVG element
+        const svgEl = DOCUMENT.createElementNS(SVG_NS, 'svg');
+        svgEl.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
+        svgEl.setAttribute('xmlns:xlink', 'http://www.w3.org/1999/xlink');
+        svgEl.setAttribute('version', '1.1');
+
+        // Add <svg> element to SVG Document
+        DOCUMENT.appendChild(svgEl);
+
+        const svgLight = lightAsset.data;
+        const svgDark = darkAsset.data;
+
+        // Parse XML from a string into a DOM Document.
+        const doc = new DOMParser();
+        const xmlLight = doc.parseFromString(svgLight, 'image/svg+xml');
+        const xmlDark = doc.parseFromString(svgDark, 'image/svg+xml');
+
+        svgEl.setAttribute('width', xmlLight.documentElement.getAttribute('width') || '');
+        svgEl.setAttribute('height', xmlLight.documentElement.getAttribute('height') || '');
+        svgEl.setAttribute('viewBox', xmlLight.documentElement.getAttribute('viewBox') || '');
+
+        // Append the root node of the DOM document to the parent element
+        await svgEl.appendChild(xmlLight.documentElement);
+        await svgEl.appendChild(xmlDark.documentElement);
+
```

Comment:
cc: @ramitha should he run svgo once more on the combined svg? probably right? to prettify it?

Reply:
I was thinking about ^ this.  I can try things out. Ty

Reply:
Update, I serialized the svgEl, then ran it through Svgo and that worked!  File is pretty! :)